syntax = "proto3";

package logprocessing;

// Define the main log entry structure
message LogEntry {
  string timestamp = 1;
  string level = 2;
  string service = 3;
  string message = 4;
  map<string, string> metadata = 5;
  string request_id = 6;
  int64 processing_time_ms = 7;
}

// Define a batch of log entries for efficient bulk processing
message LogBatch {
  repeated LogEntry entries = 1;
  string batch_id = 2;
  int64 batch_timestamp = 3;
}

// Define performance metrics
message PerformanceMetrics {
  string serialization_format = 1;
  double serialization_time_ms = 2;
  double deserialization_time_ms = 3;
  int64 data_size_bytes = 4;
  int32 entries_count = 5;
}
